src/base/settingsstorage.h

```diff
@@ -62,4 +62,34 @@ public slots:
     mutable QReadWriteLock m_lock;
 };
 
+template<typename T>
+class StorableValue
+{
+public:
+    explicit StorableValue(const QString &keyName, const QVariant &defaultValue = QVariant())
+        : m_keyName(keyName)
+        , m_defaultValue(defaultValue)
+    {
+    }
+
+    void operator= (const T &value) const
```

Comment:
> what? what fun?

We write a value into an object (which we can read back) via a const-function. Isn't is fun?

> why m_keyValue as const?

Why not? It does not change.